components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/AbstractAuthenticationDataPublisher.java

```diff
@@ -98,7 +98,7 @@ public void publishAuthenticationStepSuccess(HttpServletRequest request, Authent
         authenticationData.setForcedAuthn(context.isForceAuthenticate());
         authenticationData.setPassive(context.isPassiveAuthenticate());
         authenticationData.setInitialLogin(false);
-        authenticationData.setAuthenticator(context.getCurrentAuthenticator());
+        authenticationData.setAuthenticator((String) params.get(FrameworkConstants.AUTHENTICATOR));
```

Comment:
What if this param is not available in context ? Shall we do a check and if not found shall we add a default text which is configurable ? We already have the facility for this. Hence it should be straightforward

Reply:
Do we need that check? This will always get set by the framework now. So I think if we add that check it would be redundant.

Reply:
We can't guarantee these cases. For an example think of a scenario we are going to publish oauth related logins which are not taking place through framework. ex - password grant type. If we are to do that call through these publishers we need to check for the availability of these parameters. Otherwise those things will be shown as null or will cause errors while processing using siddhi and spark. Hence it's always better to keep it as a configurable text in a case if it's not present.

Reply:
+1 to do that.

In that case we'll need to apply the same logic for all other required parameters as well.